packages/adaptive_scaffold/lib/src/adaptive_scaffold.dart

```diff
@@ -0,0 +1,518 @@
+
+  /// Widget to be displayed in the seconaryBody slot at the smallest
+  /// breakpoint.
+  ///
+  /// If nothing is entered for this property, then the default [secondaryBody]
+  /// is displayed in the slot. If null is entered for this slot, the slot stays
+  /// empty.
+  final WidgetBuilder? largeSecondaryBody;
+
+  /// Defines the fractional ratio of body to the secondaryBody.
+  ///
+  /// For example 0.3 would mean body takes up 30% of the available space and
+  /// secondaryBody takes up the rest.
+  ///
+  /// If this value is null, the ratio is defined so that the split axis is in
+  /// the center of the screen.
+  final double? bodyRatio;
+
+  /// The breakpoint defined for the small size, associated with mobile-like
+  /// features.
+  ///
+  /// Defaults to [Breakpoints.small].
+  final Breakpoint smallBreakpoint;
+
+  /// The breakpoint defined for the medium size, associated with tablet-like
+  /// features.
+  ///
+  /// Defaults to [Breakpoints.mediumBreakpoint].
+  final Breakpoint mediumBreakpoint;
+
+  /// The breakpoint defined for the large size, associated with desktop-like
+  /// features.
+  ///
+  /// Defaults to [Breakpoints.largeBreakpoint].
+  final Breakpoint largeBreakpoint;
+
+  /// Whether or not the developer wants the smooth entering slide transition on
+  /// secondaryBody.
+  ///
+  /// Defaults to true.
+  final bool internalAnimations;
+
+  /// The orientation of the body and secondaryBody. Either horizontal (side by
+  /// side) or vertical (top to bottom).
+  ///
+  /// Defaults to Axis.horizontal.
+  final Axis bodyOrientation;
+
+  /// Whether to use a [Drawer] over a [BottomNavigationBar] when not on mobile
+  /// and Breakpoint is small.
+  ///
+  /// Defaults to true.
+  final bool useDrawer;
+
+  /// Option to override the drawerBreakpoint for the usage of [Drawer] over the
+  /// usual [BottomNavigationBar].
+  ///
+  /// Defaults to [Breakpoints.onlySmallDesktop].
+  final Breakpoint drawerBreakpoint;
+
+  /// Option to override the default [AppBar] when using drawer in desktop
+  /// small.
+  final AppBar? appBar;
+
+  /// Callback function for when the index of a [NavigationRail] changes.
+  final Function(int)? onSelectedIndexChange;
+
+  /// The width used for the internal [NavigationRail] at the medium [Breakpoint].
+  final double navigationRailWidth;
+
+  /// The width used for the internal extended [NavigationRail] at the large
+  /// [Breakpoint].
+  final double extendedNavigationRailWidth;
+
+  /// Callback function for when the index of a [NavigationRail] changes.
+  static WidgetBuilder emptyBuilder = (_) => const SizedBox();
+
+  /// Public helper method to be used for creating a [NavigationRail] from a
+  /// list of [NavigationDestination]s. Takes in a [selectedIndex] property for
+  /// the current selected item in the [NavigationRail] and [extended] for
+  /// whether the [NavigationRail] is extended or not.
+  static Builder toNavigationRail({
```

Comment:
This static method is supposed to act as a helper that sets a fixed size defined by M3 for the navigation and takes the same set of NavigationDestinations and converts them either to NavigationRailDestinations or BottomNavigationBarItems. If you think this would not really be that helpful then what I can do is make the _toRailDestination and _toBottomNavItem methods at the bottom of the adaptive_scaffold file public to replace these builders.

Reply:
If we don't think it's important for developers to be able to use these methods directly then they should be private.  If they're intended to be shared, then consider converting functions that build widgets into stateless widgets instead.

Reply:
There are some considerations to be made regarding the utility of these methods since Serena is working on improving them to match the M3 spec more. I had written these as placeholders of a sort but from what I understand and from your comments I feel like many of these may not serve much purpose.

The idea of having them as methods, though, is because we don't expect them to be used outside of the scope of the adaptive widgets suite so if we had defined them as widgets they would be accessible from anywhere after the package is imported. Having them as static methods under AdaptiveScaffold just limits the scope of use and prevents potential for misuse. If you have more feedback regarding that, please let me know. Thanks!